src/clue.js

```diff
@@ -1,14 +1,158 @@
+            age: 31,
+            description: "She is an A-list movie star with a dark past",
+            image: "https://www.radiotimes.com/uploads/images/Original/111967.jpg",
+            color: "red",
+        }
+    },
+
+    {
+        mrsPeacock: {
+            firstName: "Eleanor",
+            lastName: "Peacock",
+            occupation: "Socialité",
+            age: 36,
+            description: "She is from a wealthy family and uses her status and money to earn popularity",
+            image: "https://metrouk2.files.wordpress.com/2016/07/mrs-peacock.jpg",
+            color: "blue"
+        }
+    },
+
+    {
+        mrMustard: {
+            firstName: "Jack",
+            lastName: "Mustard",
+            occupation: "Retired Football player",
+            age: 62,
+            description: "He is a former football player who tries to get by on his former glory",
+            image: "https://static.independent.co.uk/s3fs-public/thumbnails/image/2016/07/04/08/unspecified-3.jpg",
+            color: "yellow"
+        }
+    }
+];
 
 // Rooms Collection
-const roomsArray = [];
+const roomsArray = [
+    { name: "Dining Room" },
+    { name: "Conservatory" },
+    { name: "Kitchen" },
+    { name: "Study" },
+    { name: "Library" },
+    { name: "Billiard Room" },
+    { name: "Lounge" },
+    { name: "Ballroom" },
+    { name: "Hall" },
+    { name: "Spa" },
+    { name: "Living Room" },
+    { name: "Observatory" },
+    { name: "Theater" },
+    { name: "Guest House" },
+    { name: "Patio" }
+];
 
 // Weapons Collection
-const weaponsArray = [];
+const weaponsArray = [
+    { name: "rope", weight: 10 },
+    { name: "knife", weight: 8 },
+    { name: "candlestick", weight: 2 },
+    { name: "dumbbell", weight: 30 },
+    { name: "poison", weight: 2 },
+    { name: "axe", weight: 15 },
+    { name: "bat", weight: 13 },
+    { name: "trophy", weight: 25 },
+    { name: "pistol", weight: 20 },
+]
 
 // ITERATION 2
 
+function selectRandom(array) {
+    if (array.length === 0) {
+        return undefined;
+    } else {
+        let randomNumber = Math.floor(Math.random() * Math.floor(array.length));
+        return array[randomNumber]
+    }
+}
+
+/*
+selectRandom(suspectsArray)
+selectRandom(roomsArray)
+selectRandom(weaponsArray)
+*/
+
+let mEnvelope = {}
+
+function pickMystery() {
+    mEnvelope.suspect = selectRandom(suspectsArray),
+        mEnvelope.room = selectRandom(roomsArray),
+        mEnvelope.weapon = selectRandom(weaponsArray)
+
+}
+
+pickMystery();
+console.log(mEnvelope)
+
+/*
+    function pickMystery() {
+
+        return mEnvelope = {
+            suspect: selectRandom(suspectsArray),
+            room: selectRandom(roomsArray),
+            weapon: selectRandom(weaponsArray)
+        }
+    }
+*/
 // ITERATION 3
+
+
+function revealMistery(x) {
+    console.log(`${x[0].firstName} ${x[0].last_name} killed Mr.Boddy using the ${misteryEnvelope[1].name} in the ${misteryEnvelope[2].name}!!!!`)
```

Comment:
Estuvieron cerca, solo hay que manejar mejor los arreglos y objetos.
Pero estuvieron muy cerca, pueden probar con esto :
     console.log(`${x.suspect.firstName} ${x.suspect.lastName} killed Mr. Boddy using the ${x.weapon.name} un the ${x.room.name}!`)